kpet/data.py

```diff
@@ -229,10 +230,34 @@ def matches(self, target):
         return self.__node_matches(target, True, self.data, None)
 
 
+# Removable after conversion
+def convert_sets(pattern):
```

Comment:
Might be better as a pattern method

Reply:
A cleaner and more reliable way to write this would be to simply go over every set defined in the top `index.yaml`, create a target with just that set, do a `pattern.matches()` against that target, and if it matches add the set name into the result. It won't be fast, but it's just a temporary conversion routine. As a plus, it will be simpler and easier to write correctly and verify.

That, however, means that the extraction/conversion would need to be done in the context of the `Base` object and not `Case`/`Suite` objects.

Reply:
I'm not sure how to do this without using None for the other Target parameters but using that makes any pattern with a 'not' in it not match which would ruin the idea of matching sets